torchvision/models/detection/rpn.py

```diff
@@ -466,7 +474,9 @@ def forward(self, images, features, targets=None):
         anchors = self.anchor_generator(images, features)
 
         num_images = len(anchors)
-        num_anchors_per_level = [o[0].numel() for o in objectness]
+        num_anchors_per_level_shape_tensors = [o[0].shape for o in objectness]
+        num_anchors_per_level = [s[0] * s[1] * s[2] for s in num_anchors_per_level_shape_tensors]
```

Comment:
This line will be replaced by:
        # num_anchors_per_level = [torch.prod(shape_as_tensor(o[0])) for o in objectness]

Reply:
Isn't `numel()` dynamic on the input shape?

Reply:
No, the output is an integer.

Reply:
isn't this something that we might want to change on ONNX side? Because `shape` in PyTorch currently returns a `Size` object, which is a `tuple[int]` under the hood, and using `shape` works inn ONNX

Reply:
I don't think it could be fixed in exporter. Numel is different than shape since it is traced as a constant in the IR graph. I tested this with a small repro, but I haven't looked deeper.